File: peek_plugin_search/_private/tuples/AdminStatusTuple.py
from peek_plugin_search._private.PluginNames import searchTuplePrefix
from vortex.Tuple import addTupleType, TupleField, Tuple


@addTupleType
class AdminStatusTuple(Tuple):
    __tupleType__ = searchTuplePrefix + "AdminStatusTuple"

    searchIndexCompilerQueueStatus: bool = TupleField(False)
    searchIndexCompilerQueueSize: int = TupleField(0)
    searchIndexCompilerQueueProcessedTotal: int = TupleField(0)
    searchIndexCompilerQueueLastError: str = TupleField()


    objectIndexCompilerQueueStatus: bool = TupleField(False)
    objectIndexCompilerQueueSize: int = TupleField(0)
    objectIndexCompilerQueueProcessedTotal: int = TupleField(0)
    objectIndexCompilerQueueLastError: str = TupleField()

File: peek_plugin_search/plugin-module/_private/tuples/admin/AdminStatusTuple.ts
import {addTupleType, Tuple} from "@synerty/vortexjs";
import {searchTuplePrefix} from "../../PluginNames";


@addTupleType
export class AdminStatusTuple extends Tuple {
    public static readonly tupleName = searchTuplePrefix + "AdminStatusTuple";

    searchIndexCompilerQueueStatus: boolean;
    searchIndexCompilerQueueSize: number;
    searchIndexCompilerQueueProcessedTotal: number;
    searchIndexCompilerQueueLastError: string;

    objectIndexCompilerQueueStatus: boolean;
    objectIndexCompilerQueueSize: number;
    objectIndexCompilerQueueProcessedTotal: number;
    objectIndexCompilerQueueLastError: string;

    constructor() {
        super(AdminStatusTuple.tupleName)
    }
}